### ISSUE TITLE
Use a single workfile for multiple tasks

### ISSUE THREAD

**Issue by jlorrain** [pypeclub/OpenPype#2201]
On big productions with 100+ people working on it, this become important to have a good task tracking without making the workflow for the artists too heavy.

Actually Openpype doesn't fully allow to open the same workfile and publish on different tasks. This feature would be really helpful to have more refined tasks which would be done from the same working scene.
For example, we could have a task "modeling" or "model" and a task "UV". The artist working on the UVs could directly work form the modeling scene where there is everything he needs, show his work through a review and publish his work after validation.

Overall, this would allow the production to have a more detailed tracking of the work done throughout the pipeline, more granular task attribution and easily go back and force within tasks when there are artistic retakes.

### Proposed solution

This could simply be done by linking tasks one to another represented by adding a field "master_task" to the task object of an asset within avalon's mongo.
Then the interfaces to open scene (launcher & workfiles) would just have to replace the requested task by its master task if it exist and seek for its workfiles.

That way we would have everything we need within our context for everything else to work

**Comment by tokejepsen** [comment]
Seems dangerous to have multiple people working on the same workfile at the same time. Or maybe I'm not understanding your use case?

> For example, we could have a task "modeling" or "model" and a task "UV". The artist working on the UVs could directly work form the modeling scene where there is everything he needs, show his work through a review and publish his work after validation.

Could the model task not just publish its model, then pick it up in the `UV` task?

**Comment by jlorrain** [comment]
> Seems dangerous to have multiple people working on the same workfile at the same time. Or maybe I'm not understanding your use case?

I was not refering to that kind of workflow indeed, the tasks are still intended to be done one after another and can be performed by the same person or not.

> Could the model task not just publish its model, then pick it up in the `UV` task?

Yes this would be the standard use case, the goal here is to limitate the number of workfile and scene to open when we have small tasks really close in the pipeline

**Comment by tokejepsen** [comment]
> Yes this would be the standard use case, the goal here is to limitate the number of workfile and scene to open when we have small tasks really close in the pipeline

Maybe there is a loading framework we can use here?

**Comment by tokejepsen** [comment]
> You mean a loading framework specific to each DCC ?

Just throwing ideas around atm. But instead of bending the current logic to fit these needs maybe we can use the existing publishing and loading of workfiles, and extend some functionality to fit your needs.
Could even be that the template builder is good fit here?

**Comment by mkolar** [comment]
>Yes this would be the standard use case, the goal here is to limitate the number of workfile and scene to open when we have small tasks really close in the pipeline

A few questions to clarify here. 
1. Are you publishing the model before you do UVs as well, or is the problem the fact, that you only want to really publish once the model is fully complete (possible worked on by multiple people in that time)?
2. I assume that what you're after isprettu much scenario where modeling workfile v04 automatically become the UV workfile v01.


We've seen this in animation shots where artist makes `Layout` then animator picks up this layout workfiles and uses it as a starting point for animation. The studios where we've seen it did it as simply as open the last workfile of previous task and save as v01 of the new task.

**Comment by jlorrain** [comment]
> 1. Are you publishing the model before you do UVs as well, or is the problem the fact, that you only want to really publish once the model is fully complete (possible worked on by multiple people in that time)?

Yes we publish the model before UVs, but we could need to publish it again when working on the UVs

> 2. I assume that what you're after isprettu much scenario where modeling workfile v04 automatically become the UV workfile v01.

No we can't use a serial/linear workflow here. 
Our manufacturing CG supervisors ask us to let operators work highly coupled tasks (for instance, gabarit, modeling and uv) in the same file. They also ask us to publish these in different tasks for tracking purposes.
The model is meant to be published first, and then uv; but operators must be able to rework modeling at any point after they started uv and republish it, optionally.
It's a much lighter workflow (compared to publishing the modeling and reopening / updating the uv) and it also avoids ref edits issues we had encounter with Maya.
